server/database/migrations/000004_character.sql

```diff
@@ -105,22 +106,50 @@ END
 $$;
 
 CREATE TABLE character.item(
-       name VARCHAR(32) NOT NULL,
+       name TEXT NOT NULL,
        description TEXT NOT NULL,
+       weight SMALLINT NOT NULL,
        PRIMARY KEY(name)
 );
 
 CREATE TABLE character.inventory(
-       name VARCHAR(18) NOT NULL,
+       name TEXT NOT NULL,
        e_mail TEXT NOT NULL CHECK (e_mail ~* '^[A-Za-z0-9.+%-]+@[A-Za-z0-9.-]+[.][A-Za-z]+$'),
-       username VARCHAR(32) NOT NULL,
+       username TEXT NOT NULL,
        quantity SMALLINT NOT NULL,
-       item_name VARCHAR(32) NOT NULL,
+       item_name TEXT NOT NULL,
        FOREIGN KEY (name, username, e_mail) REFERENCES character.instance(name, username, e_mail),
-       FOREIGN KEY (item_name) REFERENCES character.item(name),
-       PRIMARY KEY(name, username, e_mail, item_name, quantity)
+       FOREIGN KEY (item_name) REFERENCES character.item(name),       
+       PRIMARY KEY (name, username, e_mail, item_name)
 );
 
+CREATE OR REPLACE PROCEDURE character.update_inventory
+   (new_name TEXT, new_e_mail TEXT, new_username TEXT, new_item_name TEXT, new_quantity SMALLINT)
+   LANGUAGE plpgsql AS
+$$
+  DECLARE old_quantity SMALLINT;
+  BEGIN
+  old_quantity := NULL;
+  SELECT quantity INTO old_quantity FROM character.inventory
+    WHERE name = new_name
+    AND username = new_username
+    AND e_mail = new_e_mail
+    AND item_name = new_item_name;
+  IF old_quantity IS NOT NULL
+  THEN
+    UPDATE character.inventory
+    SET quantity = old_quantity + new_quantity
+    WHERE name = new_name
+    AND username = new_username
+    AND e_mail = new_e_mail
+    AND item_name = new_item_name;
+  ELSE
+    INSERT INTO character.inventory(name, e_mail, username, quantity, item_name)
+    VALUES (new_name, new_e_mail, new_username, new_quantity, new_item_name);
+  END IF;
```

Comment:
@EduardoLR10 @MMagueta @z-silver Wouldn't an upsert with a where clause + `EXCLUDED` replace the need for an extra `SELECT` query and be easy to get and more performant?

```sql
  INSERT INTO character.inventory(name, e_mail, username, quantity, item_name)
  VALUES (new_name, new_e_mail, new_username, new_quantity, new_item_name)
  ON CONFLICT DO 
  UPDATE SET
    -- EXCLUDED is an alias to the row that is conflicting
    -- https://www.postgresql.org/docs/17/sql-insert.html
    quantity = EXCLUDED.quantity + character.inventory.quantity,
  WHERE character.inventory.quantity IS NOT NULL;
```
here's a draft of mine

Reply:
Yes this code seems to be just an upsert. I am not sure about it being less or more performant though

Reply:
I had the impression that the original code was doing some checks with the quantity being zero or something...

Reply:
@MMagueta the only zero-based quantity check I've seen is to avoid overwriting the original quantity otherwise, I think this is easier to express with the introduction of `EXCLUDED` which is only going to be used if there's a conflict on the row. The default case will be inserting the new row with the proper quantity.